### ISSUE TITLE
Cpatured Image Taking Time To Load in Image View Via Uri

### ISSUE THREAD

**Issue by rahat14** [Dhaval2404/ImagePicker#217]
## Summary
ITs Taking a lot of time to appear in image view but gallery picked image workling fine 

## Code to reproduce
just set image uri to image view
## Android version
 11

## Impacted devices
K20 pro 

## Installation method
Maven

## SDK version
30

## Other information

https://user-images.githubusercontent.com/27925309/121653265-5266d500-cabe-11eb-86be-128b6b2bf94a.mp4

**Comment by anupdey99** [comment]
Facing the same issue, it takes significant time to preview the image.

**Comment by rahat14** [comment]
@Dhaval2404  it also now lags on android 10 on gallery options . please

**Comment by ibrotutic** [comment]
@rahat14 Are you using compression? I noticed that removing any compression from the builder helped significantly. I would guess that compression is being done on the main thread or has a memory leak, haven't looked into it.

**Comment by rahat14** [comment]
@ibrotutic  yes i am. i will remove it test it . thnx for the info

**Comment by vedraj360** [comment]
Yes compression is causing the lag. After removing it works fine.

**Comment by FauziAmth** [comment]
Yes compressing is causing significant lag.

as workaround i have to show an entire screen progress dialog when starting image picker:
```
ImagePicker.with(this)
            .compress(512)
            .createIntent {
                showProgressDialog() // show dialog here
                imagePickerResultLauncher.launch(it)
            }
```

and dismiss it when result listener called:
```
private val imagePickerResultLauncher =
        registerForActivityResult(ActivityResultContracts.StartActivityForResult()) { result ->
           // handle result
            dismissProgressDialog() // dismiss here
        }
```

**Comment by Dhaval2404** [comment]
I am looking for a possible fix.

As a workaround please use **maxResultSize** along with compress. This will speed up the compression.

```kotlin
ImagePicker.with(this)
      // Image size will be less than 512 KB
      .compress(512)
      // Image resolution will be less than 200 x 200
      .maxResultSize(200, 200)
      .createIntent {
          showProgressDialog() // show dialog here
          imagePickerResultLauncher.launch(it)
      }
```

**Comment by tayalanuj1988** [comment]
Facing the same issue

**Comment by Rishabh-More** [comment]
@Dhaval2404 The current compression method in your code isn't Optimized. If you can't resolve for a more optimized method, I suggest to take a look at this library: [Compressor](https://github.com/zetbaitsu/Compressor)

I have used this library in my previous projects, and it's compression is very optimized!
